src/hotspot/share/runtime/os.hpp

```diff
@@ -777,7 +777,7 @@ class os: AllStatic {
   static void print_register_info(outputStream* st, const void* context);
   static bool signal_sent_by_kill(const void* siginfo);
   static void print_siginfo(outputStream* st, const void* siginfo);
-  static void print_signal_handlers(outputStream* st, char* buf, size_t buflen);
+  static void print_signal_handlers(outputStream* st, char* buf, int buflen);
   static void print_date_and_time(outputStream* st, char* buf, size_t buflen);
```

Comment:
I understand you are trying to minimize changes, but these functions all used size_t consistently.  It seems odd to change the type of just this one function.

Reply:
The reason I narrowed the types is because none of the callers pass size_t and widening dll_address_to_library_name requires platform dependent changes, and also changing the Decoder::decode() functions (or adding checked_casts<> somewhere along the way) and its subtypes.
It's not simply to minimize changes, but it is.  It's to minimize casting, or checked_casting which is also not pleasant to look at in the code everywhere.  This is why I chose passing int as buflen.  There's other functions that pass int as buflen.
Also, there's no world where we pass in a buffer that's greater than 32,000.  It's used for error reporting.

Reply:
I counted (via grep) 62 occurrences of `int buflen` and 47 occurrences of `size_t` buflen in the hotspot code. So while it does look odd have two different forms right next to each other, there is at least established precedent for each.

Unfortunately these `-Wconversion` changes in general seem the antithesis of code-cleanup and consistency as they force changes in limited contexts, resulting in inconsistencies unless we let the changes fan out to extreme degrees. I would have preferred in many cases to handle overflow/underflow via assertions rather than via the primitive type system. This quest for `-Wconversion` compliance is causing a great deal of pain - hats off to @coleenp for enduring that pain to put these PRs together.

Reply:
Thank you for this comment.  This work has become soul crushing. I was optimistic at one point that we could cleanly work through these warnings by making int type consistent, and in many cases, that is possible.  And there have been some local improvements to the code that have come out of this (like primitive_compare).  But this work is hampered by 1. wanting to improve the code further than fixing the warnings, and 2. disagreements and questions about how to fix avoiding C style casts, and whether the fixes are needed, for every fix.

I'm now no longer optimistic that we'll be able to finish this.  We'll see how @dean-long does with the compiler warnings.  There are a lot more of them.  There are also a lot of warnings in the gc code.  Also, that doesn't include test code or libraries code, like java.deskop.  We need to find a way to enable/disable these on a per component basis or even a per file basis, which requires build changes I think.

When I started this around June 2, there were 424 files with 73396 warnings in hotspot (due to multiple inclusion).   After this patch, there are 259 files with 5717 warnings.

Reply:
Those statistics are absolutely impressive! You have done outstanding work on a task
that probably also feels unending and your efforts are very much appreciated.